File: 26.remove-duplicates-from-sorted-array.java
/*
 * @lc app=leetcode id=26 lang=java
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 1) {
            return 1;
        }
        int target = nums[0];
        int endpoint = nums[nums.length - 1];
        int counter = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == target) {
                nums[i] = nums[0] - 1;
                continue;
            } else if (nums[i] > target) {
                target = nums[i];
                counter++;
            }
        }
        for (int i = 0; i < nums.length; i++) {
            while (i < nums.length - 1 && nums[i] == nums[0] - 1) {
                for (int j = i; j < nums.length - 1; j++) {
                    nums[j] = nums[j + 1];
                }
            }
        }
        if (counter - 2 > 0 && nums[counter - 2] < endpoint) {
            nums[counter - 1] = endpoint;
        }

        return counter;
    }
}
// @lc code=end

File: 26.remove-duplicates-from-sorted-array.py
#
# @lc app=leetcode id=26 lang=python
#
# [26] Remove Duplicates from Sorted Array
#

# @lc code=start
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i = 1
        while i < len(nums):
            if nums[i] == nums[i-1]:
                nums.pop(i)
            else:
                i += 1
            
        
# @lc code=end

